### DISCUSSION TITLE
Can't pull image from a Git repository

### DISCUSSION THREAD

**Discussion by adicrescenzo** [containrrr/watchtower#1094] [Q&A] [answered]
Hi all,

I'm using Portainer and I've created a new Stack from a Git repository. Basically my docker-compose.yml is pulled from a Git Repo.
For this kind of container watchtower can't pull a new image from docker hub when it founds a new one, I get this error:

```
Found new nextcloud:latest image (22430e5ecdbe)
Could not do a head request for "sha256:9167b324e914d60000a13f7c2ac5ac6be60c2594b0b683a0305e63235a755374", falling back to regular pull.
Reason: registry responded to head request with "401 Unauthorized", auth: "Bearer realm=\"https://auth.docker.io/token\",service=\"registry.docker.io\",scope=\"repository:library/sha256:pull\",error=\"insufficient_scope\""
Unable to update container "/watchtower": Error response from daemon: pull access denied for sha256, repository does not exist or may require 'docker login': denied: requested access to the resource is denied. Proceeding to next.
Could not do a head request for "sha256:24a85c54a50ec36a6abc55bc699837353179675e4c2929b3e6b343ae6d758d19", falling back to regular pull.
Reason: registry responded to head request with "401 Unauthorized", auth: "Bearer realm=\"https://auth.docker.io/token\",service=\"registry.docker.io\",scope=\"repository:library/sha256:pull\",error=\"insufficient_scope\""
Unable to update container "/mosquitto": Error response from daemon: pull access denied for sha256, repository does not exist or may require 'docker login': denied: requested access to the resource is denied. Proceeding to next.
```

Can you please help me to solve this issue?
Many thanks

**Comment by piksel** [comment, marked as answer]
This is due to docker compose v2. It doesn't store the image names and therefore creates containers that cant be updated by watchtower. See the sticky issue #1050.

**Reply by adicrescenzo** [reply]
I'm pretty sure that the issue is releated to the Git repository. If I create a new stack in portainer with exacty the same content for the docker-compose.yml (v2, etc), watchtower can update the image without any issues.

**Reply by piksel** [reply]
> Could not do a head request for "sha256:9167b324e914d60000a13f7c2ac5ac6be60c2594b0b683a0305e63235a755374"

This means that the container `Image` field is set to `sha256:9167b324e914d60000a13f7c2ac5ac6be60c2594b0b683a0305e63235a755374` instead of the image name `ghcr.io/containrrr/watchtower:latest`.

I have never used portainer so, I don't know how it runs compose, but regardless, the new containers does not contain what image they were created from and hence cannot be updated.

**Reply by simonaronsson** [reply]
@adicrescenzo, sounds like it could be worth filing an issue with portainer then? If they're using the image name in a non-conforming way, they'd need to change that back. This hasn't been the case forever though, as I used to run portainer quite a lot and never had any issues in regards of compatibility with watchtower.

**Reply by adicrescenzo** [reply]
Ok guys I'll open an issue to portainer. Thanks for the clarifications
